File: 1800-concatenation-of-consecutive-binary-numbers/concatenation-of-consecutive-binary-numbers.js
var concatenatedBinary = function(n) {
    const MOD = 1_000_000_007n;
    let ans = 0n;
    for (let i = 1; i <= n; i++) {
        const nob = BigInt(Math.floor(Math.log2(i)) + 1);
        ans = ((ans << nob) | BigInt(i)) % MOD;
    }
    return Number(ans);
};

File: 1800-concatenation-of-consecutive-binary-numbers/concatenation-of-consecutive-binary-numbers.py
class Solution:
    def concatenatedBinary(self, n: int) -> int:
        MOD = 10**9 + 7
        ans = 0
        for i in range(1, n + 1):
            nob = i.bit_length()
            ans = ((ans << nob) | i) % MOD
        return ans